enclave-boundary/src/untrusted.rs

```diff
@@ -46,16 +46,42 @@ pub fn make_variable_length_ecall(
                 &mut outbuf_used,
                 &mut outbuf_retry_id,
             )
-        } {
+        };
+        match result {
             sgx_status_t::SGX_SUCCESS => match retval {
                 sgx_status_t::SGX_ERROR_OUT_OF_MEMORY => continue,
                 sgx_status_t::SGX_SUCCESS => {
                     outbuf.truncate(outbuf_used);
                     break Ok(outbuf);
                 }
+                other_retval if is_fatal_sgx_status(other_retval) => {
+                    panic!(
+                        "Enclave reported fatal error: ecall retval: {:?}",
+                        retval
```

Comment:
Should probably use the local variable name
```suggestion
                        other_retval
```